onnxruntime/core/providers/cann/npu_data_transfer.cc

```diff
@@ -67,16 +78,11 @@ common::Status NPUDataTransfer::CopyTensorAsync(const Tensor& src, Tensor& dst,
   auto& src_device = src.Location().device;
   auto& dst_device = dst.Location().device;
 
-  if (dst_device.Type() == OrtDevice::NPU) {
-    if (src_device.Type() == OrtDevice::CPU) {
-      // copy from pinned memory to NPU, this is non-blocking
-      CANN_RETURN_IF_ERROR(aclrtMemcpyAsync(dst_data,
-                                            bytes,
-                                            src_data,
-                                            bytes,
-                                            ACL_MEMCPY_HOST_TO_DEVICE,
-                                            static_cast<aclrtStream>(stream.GetHandle())));
-    } else if (src_device.Type() == OrtDevice::NPU) {
```

Comment:
Swapped the order to match the non-async method above. i.e. check NPU -> NPU first then NPU -> CPU/host accessible. 
The `else if` in the original code is misleading as there are only 3 src options: NPU, host accessible and CPU, so it was really an `else`.

This change applied in a couple of other places as well given implementations for data transfer have been copied from existing EPs when being created.